src/ui/map.js

```diff
@@ -1613,7 +1613,7 @@ class Map extends Camera {
      * @param {Object} source The source object, conforming to the
      * Mapbox Style Specification's [source definition](https://www.mapbox.com/mapbox-gl-style-spec/#sources) or
      * {@link CanvasSourceOptions}.
-     * @fires source.add
+     * @fires Map.event:source.add
```

Comment:
This event doesn't exist, unfortunately. Should this be something else?

Reply:
Adding a source triggers a whole sequence of [MapDataEvents](https://docs.mapbox.com/mapbox-gl-js/api/events/#mapdataevent) (`sourceDataType: 'content' | 'visibility' | 'metadata'`) as well as [Map Events](https://docs.mapbox.com/mapbox-gl-js/api/map/?size=n_10_n#events-data-loading) like `sourcedataloading` and `sourcedata`. But I can't really find an applicable event that's simply fired 1:1 when a source is added. I could be missing it. It sounds to me though like it's really just an echo, and if you want to know when you call addSource, maybe you should just do something yourself when you call addSource. Perhaps this ought to be removed?